activesupport/lib/active_support/number_helper/rounding_helper.rb

```diff
@@ -13,9 +13,9 @@ def round(number)
         return number unless precision
         number = convert_to_decimal(number)
         if significant && precision > 0
-          round_significant(number)
+          round_with_precision(number, precision - digit_count(number))
         else
-          round_without_significant(number)
+          round_with_precision(number, precision)
         end
```

Comment:
Can we go like this?
```
  return number unless precision

  number = convert_to_decimal(number)
  precision -= digit_count(number) if significant && precision > 0
  round_with_precision(number, precision)
```

Reply:
It's a bit confusing, since `precision` is a method, not a variable. It changes meaning, based on whether you are rounding by "number of decimals" or "significant figures".

So that code, as you've currently written it, won't quite work (it would fail with an `undefined method in nil:NilClass` error). But, I do broadly agree with your suggestion....

I've just given it another look, and found a way to delete 9 lines of code without altering the functionality 😃

Reply:
(I'm also not fully happy with [this implementation](https://github.com/rails/rails/pull/38148/files#diff-b376ee796fb7057e9152f3ae7832767cR23-R37), for similar reasons. I'd welcome someone taking a quick go at simplifying that, too! 😅)